api/v1beta1/awsmachine_types.go

```diff
@@ -157,6 +157,11 @@ type AWSMachineSpec struct {
 	// +optional
 	// +kubebuilder:validation:Enum:=default;dedicated;host
 	Tenancy string `json:"tenancy,omitempty"`
+
+	// CapacityReservationID specifies the instance that should be launched in the
+	// reserved compute capacity.
+	// +optional
+	CapacityReservationID string `json:"capacityReservationId,omitempty"`
```

Comment:
What's the motivation to add this field to v1beta1? In general, new fields should probably only go into v1beta2.

Reply:
I added the same based on the spot market options PR. I will remove that.